### PR TITLE
Overhaul: Simplify proxy, additional-variable, process-variable

### PR DESCRIPTION
**Pull request by andreasebner**

### CODE REVIEW

File: src/ua_proxies.cpp [hunk 1]
```diff
@@ -27,14 +27,15 @@
 using namespace std;
 
 UA_NodeId* nodePairList_getTargetIdBySourceId(nodePairList pairList, UA_NodeId remoteId) {
-  UA_NodeId* local = nullptr;
+/*  UA_NodeId* local = nullptr;
```
Comment (zenker) [inline src/ua_proxies.cpp:30]: Can we remove this function at all?
It is not used....

Comment (andreasebner) [reply src/ua_proxies.cpp:30]: Currently there is no further use planned. I will update the PR and remove the function.

**Comment by zenker** [comment]
All additional nodes added to a chimeraTK PV are of type BaseDataType (e.g. Description, EngineeringUnit, Name....).
Like that I can not read them with LabView as before because there is a type mismatch. Please set proper data types for them - mostly string except for Validity which is uint?
Probably - I did not test that -  this also applies to additional variables that can be added in the mapping process  - they should also be of type string.

**Comment by zenker** [comment]
All node are read only!
Somehow the write access was dropped.

**Comment by andreasebner** [comment]
> All additional nodes added to a chimeraTK PV are of type BaseDataType (e.g. Description, EngineeringUnit, Name....). Like that I can not read them with LabView as before because there is a type mismatch. Please set proper data types for them - mostly string except for Validity which is uint? Probably - I did not test that - this also applies to additional variables that can be added in the mapping process - they should also be of type string.

Could you give some more details which application is used? If i run the contained executables everything looks fine to me: 

Screenshot 2023-04-27 085321

**Comment by zenker** [comment]
That's interesting - if I drag the PVs to the data access view the type is indeed correct. But if you check the attributes part you will see that the DataType identifier is 24 [BaseDataTytpe].
image

**Comment by andreasebner** [comment]
> All node are read only! Somehow the write access was dropped.

This issue could be reporoduced and is fixed.

**Comment by andreasebner** [comment]
> That's interesting - if I drag the PVs to the data access view the type is indeed correct. But if you check the attributes part you will see that the DataType identifier is 24 [BaseDataTytpe]. image

Thanks for the hint. DataType was transmitted correctly but not set in the node itself. I pushed a commit for solving this issue.

**Comment by zenker** [comment]
Now I get quite a lot messages like:
`[2023-04-27 09:58:59.111 (UTC+0200)] info/session	SecureChannel 0 | Session "Administrator" | AddNode (ns=1;s=watchdog_server/Configuration/Configuration/MicroDAQ/decimationFactor/EngineeringUnit): The value is empty. But this is only allowed for BaseDataType. Create a matching default value.`

Can we set a default - like empty string and data valid?

File: src/csa_processvariable.cpp [hunk 2]
```diff
@@ -1741,43 +1348,59 @@ UA_StatusCode ua_processvariable::addPVChildNodes(UA_NodeId pvNodeId, string bas
...
-  //Adding the EngineeringUnit node to the PV
+  //Adding the 'EngineeringUnit' node to the PV
+  UA_DataSource_Map_Element mapElemEU;
+  mapElemEU.typeTemplateId = UA_NODEID_NUMERIC(CSA_NSID, CSA_NSID_VARIABLE_UNIT);
+  mapElemEU.description = UA_LOCALIZEDTEXT((char*)"", (char*)"");
+  mapElemEU.dataSource.read = ua_readproxy_ua_processvariable_getEngineeringUnit;
+  mapElemEU.dataSource.write = NULL;
   UA_VariableAttributes_init(&attr);
   attr = UA_VariableAttributes_default;
   attr.displayName = UA_LOCALIZEDTEXT((char*)"en_US", (char*)"EngineeringUnit");
```
Comment (zenker) [inline src/csa_processvariable.cpp:1368]: Here you use "en_US" but for all other additional nodes you use "".
Please be consistent and add "en_US" everwhere...

Comment (andreasebner) [reply src/csa_processvariable.cpp:1368]: Solved.

Comment (andreasebner) [reply src/csa_processvariable.cpp:1368]: > Now I get quite a lot messages like: `[2023-04-27 09:58:59.111 (UTC+0200)] info/session SecureChannel 0 | Session "Administrator" | AddNode (ns=1;s=watchdog_server/Configuration/Configuration/MicroDAQ/decimationFactor/EngineeringUnit): The value is empty. But this is only allowed for BaseDataType. Create a matching default value.`
> 
> Can we set a default - like empty string and data valid?

Done for Validity.

**Pull request merged**